gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -22,63 +24,88 @@ def __init__(self, service):
                 ["read", "write", "notify"], service)
         self.add_descriptor(WifiConnectDescriptor(self))
         self.add_descriptor(OpaqueStructureDescriptor(self))
-        self.wifi_status = ""
 
-    def WiFiConnectCallback(self):
+        # Credential to connect to a WiFi
+        self.wifi_service = ""
+        self.wifi_password = ""
+
+    def check_wifi_status(self):
+        # Check the current wi-fi connection status
+        logger.debug('Check WiFi Connect')
+        nm_state = str(nmcli_custom.device.show('wlan0')['GENERAL.STATE'].split(" ")[0])
+
+        # Convert the network manager device state into wifi status response
+        if nm_state == constants.NM_DEVICE_STATE_ACTIVATED:
+            wifi_status = constants.WIFI_CONNECTED
+        elif nm_state == constants.NM_DEVICE_STATE_FAILED:
+            wifi_status = constants.WIFI_ERROR
+        else:
+            wifi_status = constants.WIFI_INVALID_PASSWORD
```

Comment:
I still find this logic suspect. If the network manager isn't activated or failed, then we consider it to have a bad password. But there are a bunch of possible NM states. Do we really want to treat all of those as invalid passwords? 

I'm also still unclear how the mobile app error message is being generated. Can you point me to the relevant code please?

Reply:
What I can get as the result of connecting is CONNECTED/DISCONNECTED.
```
        if nm_state == constants.NM_DEVICE_STATE_ACTIVATED:
            wifi_status = constants.WIFI_CONNECTED
        elif nm_state == constants.WHATEVER_IS_FOR_INVALID_PASSWORD:
            wifi_status = constants.WIFI_INVALID_PASSWORD
        else:
            wifi_status = constants.WIFI_ERROR
```
The actual value returning for constants.WHATEVER_IS_FOR_INVALID_PASSWORD is constants.NM_DEVICE_DISCONNECTED.

I will refactor `constants.NM_DEVICE_DISCONNECTED` mapped into `constants.WIFI_INVALID_PASSWORD` to make more sense.